src/Microsoft.DiaSymReader.PortablePdb/SymDocument.cs

```diff
@@ -152,9 +200,24 @@ public int GetSourceLength(out int length)
 
         public int HasEmbeddedSource(out bool value)
         {
-            // SymReader doesn't support embedded source.
-            value = false;
-            return HResult.E_NOTIMPL;
+            int hr;
+            int length;
+            hr = GetSourceLength(out length);
+            value = length > 0;
+            return hr == HResult.S_FALSE ? HResult.S_OK : hr;
+        }
+
+        private int GetEmbeddedSourceBlobReader(out BlobReader reader)
```

Comment:
I think it would be better to directly return BlobReader here and let the callers decide on how to handle ```default(BlobReader)```